src/App.tsx

```diff
@@ -1,61 +1,113 @@
+import { FormEventHandler, useState } from 'react';
 import './App.scss';
 
-// import usersFromServer from './api/users';
-// import todosFromServer from './api/todos';
+import usersFromServer from './api/users';
+import todosFromServer from './api/todos';
+import { TodoList } from './components/TodoList';
+import { Task } from './react-app-env';
+
+function getNextTaskId(tasks:Task[]) {
+  return tasks.sort((task1, task2) => task2.id - task1.id)[0].id + 1;
+}
 
 export const App = () => {
+  const [input, setInput] = useState('');
+  const [inputError, setInputError] = useState(false);
+  const [select, setSelect] = useState('');
+  const [selectError, setSelectError] = useState(false);
+  const [todos, setTodos] = useState<Task[]>(todosFromServer.map(todo => ({
+    id: todo.id,
+    title: todo.title,
+    completed: todo.completed,
+    user: usersFromServer.find(user => user.id === todo.userId) || null,
+  })));
+
+  const handleSubmit: FormEventHandler = event => {
+    event.preventDefault();
+    // eslint-disable-next-line no-console
+    console.log(input, select);
+
+    if (!input) {
+      setInputError(true);
+    } else {
+      setInputError(false);
+    }
+
+    if (!select) {
+      setSelectError(true);
+    } else {
+      setSelectError(false);
+    }
+
+    if (input && select) {
+      const task: Task = {
+        id: getNextTaskId(todos),
+        title: input,
+        completed: false,
+        user: usersFromServer.find(user => user.name === select) || null,
+      };
+
+      if (task !== undefined) {
+        setTodos(oldTodos => [...oldTodos, task]);
+      }
+
+      setSelect('');
+      setInput('');
+    }
+  };
+
   return (
     <div className="App">
       <h1>Add todo form</h1>
 
-      <form action="/api/users" method="POST">
+      <form
+        action="/api/users"
+        method="POST"
+        onSubmit={event => handleSubmit(event)}
```

Comment:
You already pass event in `handleSubmit` 
```suggestion
        onSubmit={handleSubmit}
```